### DISCUSSION TITLE
Find themes/plugins results in a segmentation fault.

### DISCUSSION THREAD

**Discussion by buckybytes** [e107inc/e107#5184] [Q&A Core] [answered]
Whenever I click on Find Themes or Find Plugins, I get an error.  Something this drastic leads me to believe I have missed an installation step, or some other "non-bug" explanation.  Anybody have an idea of where to start digging?  My google-fu is lacking on this one.

In chrome, it appears as ERR_NO_RESPONSE, but if I check the Apache error log, I see a child process has seg-faulted at the timestamps I make the requests.  I'm running a dedicated Debian 11 server with just a LAMP stack.  

Info dump:

=============================================
e107
Version 2.3.3

Security level
[5] Balanced

Site Theme
bootstrap5

Admin Theme
Bootstrap 3 v1.0 by e107 Inc (2013-12-25)

Install date
Sunday 26 November 2023 - 20:33

Server
Apache/2.4.54 (Debian)
(host: mysite.com)

PHP Version
8.2.0

MySQL
8.0.31
Database: e107
PDO: Enabled
Mode: NO_ENGINE_SUBSTITUTION

Charset
utf-8

**Comment by Deltik** [comment, marked as answer]
e107 was previously affected by a segmentation fault bug in PHP 8.2.0 (https://github.com/php/php-src/issues/10200), but we applied a workaround in https://github.com/e107inc/e107/commit/6b75f341ad54bdc8c7333acecd0ca435a07b2086 that is in e107 v2.3.3.

PHP 8.2.0 is not an official version included with Debian 11 (Debian 11 had PHP 7.4.21 while Debian 12 had PHP 8.2.5).  Can you upgrade it to PHP 8.2.15 and see if the segmentation fault still happens?

If not, can you compile PHP with the `--enable-debug` option to see if you get any assertion failures?

You can also hook up [Xdebug](https://xdebug.org/) and step through every line of code until you encounter the one that crashes PHP.

Yet another way is to obtain a [core dump](https://stackoverflow.com/a/67989782) from the segmentation fault and [generate a backtrace](https://bugs.php.net/bugs-generating-backtrace.php) from that core dump.

**Reply by buckybytes** [reply]
8.2.15 did the trick for both plugins and themes.  Thanks!
